### ISSUE TITLE
What is the best way of defining optional event handlers in propTypes?

### ISSUE THREAD

**Issue by alisabzevari** [yannickcr/eslint-plugin-react#1067]
I have a component that has an `onChange` event. Providing the event handler for this component is optional. I have defined the `propTypes` as follows:
``` js
Cmp.propTypes = {
  onChange: React.PropTypes.func
}
```
But I will get the `react/require-default-props` eslint error.
My question is that how can I define a propType of `Function | undefined`?

**Comment by petersendidit** [comment]
The rule is checking if there is a default prop for non required props.

You can set a noop function as the default:
```js
Cmp.defaultProps = {
    onChange: () => {}
}
```

**Comment by ljharb** [comment]
You can also set `null` as the default.

**Comment by wyattdanger** [comment]
@ljharb is that behavior still true? i'm running into an issue today that I didn't expect, where I have a PropType of node and a default of null and am getting proptype warnings

**Comment by joelnbl** [comment]
> You can also set `null` or `undefined` as the default, since all non-required props allow those.
Important thing to keep in mind is that an empty function ()=>{} is an object hence its truthy whereas null and undefined are falsy.

So if you want to check whether that event handler prop is passed on implementation and handle both cases (supplied and not supplied) differently, it makes more sense to go with null or undefined.

reference: https://stackoverflow.com/questions/54025892/reactjs-defaultprops-empty-function-declaration
